### PR TITLE
fix(bundler) fix pretty path resolution

### PR DESCRIPTION
**Pull request by cirospaciari**
### What does this PR do?
Fix:  https://github.com/oven-sh/bun/issues/15103
Fix: https://github.com/oven-sh/bun/issues/14835
Fix: https://github.com/oven-sh/bun/issues/14688
Fix: https://github.com/oven-sh/bun/issues/14590
Fix: https://github.com/oven-sh/bun/issues/13897

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?
Tests

### CODE REVIEW

**Comment by robobun** [comment]
@cirospaciari, your commit b6adf8776a8a7288946b622cc4d9fee4d5509958 is building: [`#5940`](https://buildkite.com/bun/bun/builds/5940)

File: src/bundler/bundle_v2.zig [hunk 1]
```diff
@@ -9581,13 +9577,19 @@ pub const LinkerContext = struct {
...
-                const source: Logger.Source = sources[part_range.source_index.get()];
+                const source: *Logger.Source = &sources[part_range.source_index.get()];
 
                 const file_path = brk: {
                     if (source.path.isFile()) {
                         // Use the pretty path as the file name since it should be platform-
                         // independent (relative paths and the "/" path separator)
+                        const path = source.path;
+                        if (path.pretty.ptr == path.text.ptr) {
+                            const rel = bun.path.relativePlatform(c.resolver.fs.top_level_dir, path.text, .loose, false);
```
Comment (paperclover) [inline src/bundler/bundle_v2.zig]: probably should use `pathWithPrettyInitialized` to handle an edgecase in dev server

Comment (cirospaciari) [reply src/bundler/bundle_v2.zig]: I dont think I have a `*BundleV2` ref to use here, let me check

File: src/bundler/bundle_v2.zig [hunk 2]
```diff
@@ -9581,13 +9577,19 @@ pub const LinkerContext = struct {
...
             const sources = c.parse_graph.input_files.items(.source);
             for (chunk.content.javascript.parts_in_chunk_in_order) |part_range| {
-                const source: Logger.Source = sources[part_range.source_index.get()];
+                const source: *Logger.Source = &sources[part_range.source_index.get()];
 
                 const file_path = brk: {
                     if (source.path.isFile()) {
                         // Use the pretty path as the file name since it should be platform-
                         // independent (relative paths and the "/" path separator)
+                        const path = source.path;
```
Comment (Jarred-Sumner) [inline src/bundler/bundle_v2.zig]: Is copying this to the stack intentional?

Comment (cirospaciari) [reply src/bundler/bundle_v2.zig]: source is a ptr so avoiding accessing it multiple times, but I can just remove this if copying is not cheaper than the accesses.

File: src/bundler/bundle_v2.zig (hunk 1 above)
Comment (cirospaciari) [reply src/bundler/bundle_v2.zig]: Ok centralized this logic in one place now, should handle the dev server edgecase.

**Pull request merged**